ios/browser/api/url_sanitizer/url_sanitizer_service.mm

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/url_sanitizer/browser/url_sanitizer_service.h"
+
+#include <string>
+#include "brave/ios/browser/api/url_sanitizer/url_sanitizer_service+private.h"
+#import "net/base/mac/url_conversions.h"
+#include "url/gurl.h"
+
+@interface URLSanitizerService () {
+  brave::URLSanitizerService* urlSanitizer_;
+}
+
+@end
+
+@implementation URLSanitizerService
+
+- (instancetype)initWithURLSanitizerService:
+    (brave::URLSanitizerService*)urlSanitizer {
+  self = [super init];
+  if (self) {
+    urlSanitizer_ = urlSanitizer;
+  }
+  return self;
+}
+
+- (NSURL*)sanitizedURL:(NSURL*)url {
+  GURL gurl = net::GURLWithNSURL(url);
```

Comment:
`DCHECK(urlSanitizer_)` should be on the first line of this function in case that `urlSanitizer_` pointer is invalid.

Reply:
Done